pkg/controller/backupentry/actuator.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backupentry
+
+import (
+	"context"
+	"strings"
+
+	controller "github.com/gardener/gardener-extensions/pkg/controller"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/go-logr/logr"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+)
+
+const (
+	// EtcdBackupSecretName is the name of secret having credentials for etcd backups.
+	EtcdBackupSecretName string = "etcd-backup"
+
+	backupBucketName string = "bucketName"
+)
+
+// Actuator acts upon BackupEntry resources.
+type Actuator interface {
+	// Reconcile reconciles the BackupEntry.
+	Reconcile(context.Context, *extensionsv1alpha1.BackupEntry) error
+	// Delete deletes the BackupEntry.
+	Delete(context.Context, *extensionsv1alpha1.BackupEntry) error
+}
+
+// ProviderActuator preforms provider specific operation with BackupBucket resources.
+type ProviderActuator interface {
+	// Delete deletes the BackupBucket.
+	Delete(context.Context, *extensionsv1alpha1.BackupEntry) error
+	// GetETCDSecretData return the updated the secret data as per provider requirement.
+	GetETCDSecretData(context.Context, *extensionsv1alpha1.BackupEntry, map[string][]byte) (map[string][]byte, error)
+}
+
+type actuator struct {
```

Comment:
Similar to `pkg/worker/genericactuator` or `pkg/controlplane/genericactuator` can you please move the `providerActuator` into a separate package? It might also makes sense thinking about renaming it to `genericactuator`.

Reply:
Sure.